### PR TITLE
Bug fix to support ARM template deployments

### PR DESCRIPTION
**Pull request by harshivcodes**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [x] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
       - Reviewed the breaking change doc, since we are adding and removing a response code. It sounds like it will be a breaking change. Will schedule a meeting with the Breaking Change Review board.
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [x] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [x] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews]
- [x] Adding and removing a response code from PUT APIs for 2 resources.
- [ ] (https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/storagepool/resource-manager/Microsoft.StoragePool/preview/2020-03-15-preview/storagepool.json [hunk 1]
```diff
@@ -195,7 +195,7 @@
               "$ref": "#/definitions/DiskPool"
             }
           },
-          "202": {
+          "201": {
```
Comment (Juliehzl) [inline specification/storagepool/resource-manager/Microsoft.StoragePool/preview/2020-03-15-preview/storagepool.json:198]: In my test, your server is still returning 202, but here you change to 201, which will cause misunderstanding for SDK.

Comment (RamyasreeChakka) [inline specification/storagepool/resource-manager/Microsoft.StoragePool/preview/2020-03-15-preview/storagepool.json:198]: As per this spec, x-ms-long-running-operation is set to true for PUT request which means its a long running operation. Can you specify the x-ms-long-running-operation-options as well to let the callers know the long operation pattern.
Refer http://azure.github.io/autorest/extensions/#x-ms-long-running-operation-options

Also doesn't your service return 202 status now?

**Comment by Juliehzl** [comment]
@harshitha-akkaraju could you fix comments above first?

**Comment by chiragg4u** [comment]
Please go through the contribution checklist added as part of the review and mark them completed. This is a requirement before review can be continued.

File: specification/storagepool/resource-manager/Microsoft.StoragePool/preview/2020-03-15-preview/storagepool.json (hunk 1 above)
Comment (harshitha-akkaraju) [reply specification/storagepool/resource-manager/Microsoft.StoragePool/preview/2020-03-15-preview/storagepool.json:198]: @Juliehzl we were waiting for the private builds of the PSH module and az cli to be updated with the new status code before we make the code change.

The server should be returning the right status code now.

Comment (harshitha-akkaraju) [reply specification/storagepool/resource-manager/Microsoft.StoragePool/preview/2020-03-15-preview/storagepool.json:198]: @RamyasreeChakka fixed this.

Yes, at the time we were returning 202, but now the server should be returning 201.

**Review by j5lim — APPROVED** [review]

**Review by JeffreyRichter — APPROVED** [review]
No customers yet

**Review by Juliehzl — APPROVED** [review]

**Pull request merged**